azext_iot/central/services/device.py

```diff
@@ -97,9 +97,13 @@ def list_devices(
     query_parameters = {}
     query_parameters["api-version"] = api_version
 
+    logger.warning(
+        "This command may take a long time to complete if your app contains a lot of devices"
+    )
+
     pages_processed = 0
-    while (pages_processed <= max_pages) and url:
-        response = requests.get(url, headers=headers, params=query_parameters)
+    while (max_pages == 0 or pages_processed <= max_pages) and url:
```

Comment:
Should this be pages_processed < max_pages?

let's say we have max_pages = 1
We start with pages_processed at 0, so we get and process that page, increase the count of pages processed.
Now that pages_processed is 1, we should not process another page (but the code says we will).

Reply:
good catch. fixed here and in other places with same bug